lib/services/auth.dart

```diff
@@ -50,23 +52,35 @@ class Auth {
   }
 
   Future<bool> checkPINExists() async {
-    String pin = await getCurrentUserPIN();
-    return pin.length == 4 &&
-        num.tryParse(pin) != null; // unless PIN is 4 digits, return false
+    if (await getCurrentUser() != null) {
+      Admin admin = Admin(user: auth.currentUser);
+      String pin =
+          (!mock) ? await getCurrentUserPIN() : await admin.getCurrentUserPIN();
+      return pin.length == 4 &&
+          num.tryParse(pin) != null; // unless PIN is 4 digits, return false
+    } else {
+      return false;
+    }
   }
 
   Future<String> createPIN(String pin) async {
     if (await getCurrentUser() != null) {
       try {
         if (pin.length == 4 && num.tryParse(pin) != null) {
-          final docUser =
-              FirebaseFirestore.instance.collection('userPins').doc();
-          final entry = {
-            //pin is saved with user's ID
-            'pin': pin,
-            'userId': await getCurrentUserId(),
-          };
-          await docUser.set(entry); //adds PIN to database
+          if (!mock) {
```

Comment:
why checking if not mock here. Again, no method should check for that. The only place in the code where this is possible is to prevent a loading indicator from appearing during a test. Please remove ALL the if (isMock) in your code and replace them by proper mocking in your tests